### PR TITLE
AMS-6006 added topstory, priority & mod_pubdate into push to redfact

### PR DESCRIPTION
**Pull request by leighfwarren**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -1,17 +1,20 @@
...
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
 
   <modelVersion>4.0.0</modelVersion>
   <name>Camel Post To Redfact</name>
   <description>Camel Post To Redfact Plugin</description>
   <groupId>com.atex.plugins</groupId>
   <artifactId>camel-post-to-redfact</artifactId>
-  <version>1.8-SNAPSHOT</version>
+  <version>1.0-SNAPSHOT</version>
```
Comment (prabjohns-atex) [inline pom.xml:11]: why is this 1.0 ?

File: pom.xml [hunk 2]
```diff
@@ -1,17 +1,20 @@
...
   <packaging>jar</packaging>
 
   <parent>
     <groupId>com.polopoly</groupId>
     <artifactId>public</artifactId>
-    <version>10.20.2</version>
+    <version>11.0.0-fp2</version>
```
Comment (prabjohns-atex) [inline pom.xml:17]: why change this ?

File: pom.xml [hunk 3]
```diff
@@ -67,31 +72,17 @@
   </scm>
 
   <properties>
-    <polopoly.version>10.20.2</polopoly.version>
+    <polopoly.version>11.0.0-fp2</polopoly.version>
```
Comment (prabjohns-atex) [inline pom.xml:75]: this will be a problem too

File: pom.xml [hunk 4]
```diff
@@ -292,7 +250,7 @@
         <plugins>
           <plugin>
             <artifactId>maven-failsafe-plugin</artifactId>
-            <version>2.19</version>
+            <version>2.6</version>
```
Comment (prabjohns-atex) [inline pom.xml:253]: why change this ?

File: src/main/java/com/atex/onecms/app/dam/integration/camel/component/redfact/RedFactUtils.java [hunk 5]
```diff
@@ -130,7 +130,7 @@ public RedFactFormArticle convert(String imageServiceUrl, RedfactConfig redFactC
         return new RedFactFormArticle(source.getContentId().getContentId().toString(), formArticleParams, redFactFormImages);
     }
 
-    public List<NameValuePair> convertArticleToForm(ContentResult<OneArticleBean> source, OneArticleBean damArticle) {
+    public List<NameValuePair> convertArticleToForm(ContentResult<CustomArticleBean> source, CustomArticleBean damArticle) {
```
Comment (prabjohns-atex) [inline src/main/java/com/atex/onecms/app/dam/integration/camel/component/redfact/RedFactUtils.java:133]: can we use reflection instead ? Adding a dependancy on adm-binary will cause a problem.

File: src/main/java/com/atex/onecms/app/dam/integration/camel/component/redfact/RedFactUtils.java [hunk 6]
```diff
@@ -158,17 +158,17 @@ public List<NameValuePair> convertArticleToForm(ContentResult<OneArticleBean> so
...
         int priority = damArticle.getPriority();
-        String topStory = "0";
-        if (priority == 0)
-            priority = calculatePriority(damArticle.getName(), damArticle.getWords());
-        else {
-            if (priority > 10) {
-                priority = priority - 10;
-                topStory = "1";
+        params.add(new BasicNameValuePair("priority", Integer.toString(priority)));
+        Map<String, Map<String,String>> propertyBag = damArticle.getPropertyBag();
```
Comment (prabjohns-atex) [inline src/main/java/com/atex/onecms/app/dam/integration/camel/component/redfact/RedFactUtils.java:162]: get via reflection

**Review by prabjohns-atex — CHANGES_REQUESTED** [review]
Adding a dependency on adm binary will cause circular dependency issues at some point. Would be better to use reflection to get what you need.

File: src/main/java/com/atex/onecms/app/dam/integration/camel/component/redfact/RedFactUtils.java (hunk 5 above)
Comment (leighfwarren) [reply src/main/java/com/atex/onecms/app/dam/integration/camel/component/redfact/RedFactUtils.java]: i have used reflection in the new commit

File: src/main/java/com/atex/onecms/app/dam/integration/camel/component/redfact/RedFactUtils.java (hunk 6 above)
Comment (leighfwarren) [reply src/main/java/com/atex/onecms/app/dam/integration/camel/component/redfact/RedFactUtils.java]: done

**Pull request closed without merging**